### ISSUE TITLE
Using custom type handler for objects inside array/hash refs

### ISSUE THREAD

**Issue by ryan-lang** [moose/MooseX-Storage#10]
The docs mention that arrayrefs and hashrefs will be inspected up to one level down to serialize/deserialize any objects that are found, which is great.  However, it appears that rather than checking types against the main types list, any objects are passed directly to the generic "object handler".  This means that any custom types added with `add_custom_type_handler` are ignored in this scenario.

Can anyone shed some light on whether this is by-design, a potential area of improvement, or if there's a good workaround?

**Comment by ryan-lang** [comment]
Specifically, the code in question:
https://metacpan.org/source/ETHER/MooseX-Storage-0.49/lib/MooseX/Storage/Engine.pm#L279
(MooseX::Storage::Engine, Lines 279 - 281)

**Comment by karenetheridge** [comment]
Ryan, is this a regression in 0.49? The logic with deep objects has
recently been enhanced.

**Comment by ryan-lang** [comment]
No, the code in question seems to have existed in at least the last couple versions.

Comparing them now, I see that 0.49 added some new logic nearby, but didn't alter the part that sends all blessed objects to the generic handler (not checked for custom type handlers).
